tsfc/driver.py

```diff
@@ -349,3 +354,12 @@ def lower_integral_type(fiat_cell, integral_type):
         entity_ids = list(range(len(fiat_cell.get_topology()[integration_dim])))
 
     return integration_dim, entity_ids
+
+
+def pick_mode(mode):
+    "Return one of the specialized optimisation modules from a mode string."
+    if mode == "vanilla":
+        import tsfc.vanilla as m
```

Comment:
How do you (or do you?) envisage composing modes?  For example, it may well be beneficial for "optimised" modes to unroll index sums too.  Would they themselves import the vanilla mode and use it?